src/crimson/common/errorator.h

```diff
@@ -0,0 +1,827 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <exception>
+#include <seastar/core/future-util.hh>
+
+namespace ceph {
+
+// TODO: consider specializing the templates instead of `plainify`-ing.
+template<class Container, class AsyncAction>
+static inline auto do_for_each(Container& c, AsyncAction action) {
```

Comment:
this scares me. do we always need the boxing/unboxing between `seastar::future<>` and `ceph::future<>` explictly for the type-erasured error handling?